include/LightGBM/config.h

```diff
@@ -712,6 +712,11 @@ struct Config {
   // desc = **Note**: can be used only in CLI version; for language-specific packages you can use the correspondent function
   bool save_binary = false;
 
+  // [no-save]
```

Comment:
My initial motivation for this PR is to ease comparision of prediction result with cmdline and other language (e.g. python). So I didn't thought about saving this parameter in the model file.

To be more specific

1. the model is trained with cmdline
2. prediction is done with both python and cmdline
    - this requires precise float parsing in order to make both prediction have the same result

As this option can also be used for training, I think it's better to include it in the model file. At least this can add information about what float parsing mode is used during train.

I'll remove `no-save`  here.